mlflow/models/docker_utils.py

```diff
@@ -84,6 +88,10 @@ def generate_dockerfile(
     install_mlflow_steps = _pip_mlflow_install_step(output_dir, mlflow_home)
 
     if base_image.startswith("python:"):
+        if install_java is not None:
```

Comment:
```suggestion
        if install_java:
```
Since we don't install java when it's False anyways, the warning will make users wonder if java is installed or not